WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsUtils.java

```diff
@@ -176,6 +177,11 @@ public static void refreshMetadata(AccountStore accountStore, SiteStore siteStor
         metadata.setNumBlogs(siteStore.getSitesCount());
         metadata.setUsername(accountStore.getAccount().getUserName());
         metadata.setEmail(accountStore.getAccount().getEmail());
+        if (BuildConfig.BUILD_TYPE == "debug") {
```

Comment:
I am not 100% sure what is the correct flag to use here, but for consistency I'd do the following instead:

```suggestion
        if (BuildConfig.DEBUG) {
```

I believe this is the most common usage in the app. Of course if we do this, we probably wouldn't want to set the scheme to `BuildConfig.BUILD_TYPE` but instead to "debug" directly 🤷‍♂️ 

If you want to use the `BUILD_TYPE` property instead, which is also very reasonable as they are set [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/build.gradle#L136-L148), then I'd suggest changing it to:

```
if ("debug".equalsIgnoreCase(BuildConfig.BUILD_TYPE)) {
```

Kotlin spoiled us as we still need to use `equals` for comparisons in Java. (AS will also complain about this) I don't like this too much because it uses a string directly instead of from a constant, but I don't think there is a constant we can read here. (and defining a new one doesn't improve it that much)

---

An alternative might be to combine the flavor and the build type as Android Studio does (`vanillaRelease`, `vanillaDebug` etc) or send a separate property for the build type and keep the `scheme` to just flavor. However, that may have consistency issues across platforms.

_P.S: Sorry for the long comment for a single line, just wanted to offer different possibilities so you can choose whichever you like the most. Also, `scheme` name is very iOSy but I understand it is used for consistency._

Reply:
`BuildConfig.DEBUG` is much nicer, thanks for suggesting it! 
> Of course if we do this, we probably wouldn't want to set the scheme to BuildConfig.BUILD_TYPE but instead to "debug" directly

Makes sense to me. I'll make this change. 

> I also tested this with breakpoints instead of the live tracker

Totally fine.  The tracker is slower. Its nice for an end-to-end check, but not really necessary.

Thank you Oguz! I appreciate the options and explanations :)

Reply:
> Also, scheme name is very iOSy but I understand it is used for consistency

This is a fair point.  Android adds a `user_info_` prefix to global properties and iOS does not so there is already some inconsistency in the naming.  We could go with something different.  I like "scheme" because it does hint at the , cross platform usage. I thought something like `buildType` or `variant` might be better but there could be confusion with the similarly named `BuildConfig` constants.  Open to other ideas if there's a better option. :)